ros_controls.rolling-on-humble.repos

```diff
@@ -27,11 +27,79 @@ repositories:
     type: git
     url: https://github.com/ros-controls/ros2_control.git
     version: master
-  # ros-controls/ros2_control_demos:
-  #   type: git
-  #   url: https://github.com/ros-controls/ros2_control_demos.git
-  #   version: master
+  ros-controls/ros2_control_demos:
+    type: git
+    url: https://github.com/ros-controls/ros2_control_demos.git
+    version: master
   ros-controls/ros2_controllers:
     type: git
     url: https://github.com/ros-controls/ros2_controllers.git
     version: master
+  gz/gz_plugin_vendor:
+    type: git
+    url: https://github.com/gazebo-release/gz_plugin_vendor.git
+    version: rolling
+  gz/gz_sim_vendor:
+    type: git
+    url: https://github.com/gazebo-release/gz_sim_vendor.git
+    version: rolling
+  gz/gz_cmake_vendor:
+    type: git
+    url: https://github.com/gazebo-release/gz_cmake_vendor.git
+    version: rolling
+  gz/gz_utils_vendor:
+    type: git
+    url: https://github.com/gazebo-release/gz_utils_vendor.git
+    version: rolling
+  gz/gz_tools_vendor:
+    type: git
+    url: https://github.com/gazebo-release/gz_tools_vendor.git
+    version: rolling
+  gz/sdformat_vendor:
+    type: git
+    url: https://github.com/gazebo-release/sdformat_vendor.git
+    version: rolling
+  gz/gz_math_vendor:
+    type: git
+    url: https://github.com/gazebo-release/gz_math_vendor.git
+    version: rolling
+  gz/gz_transport_vendor:
+    type: git
+    url: https://github.com/gazebo-release/gz_transport_vendor.git
+    version: rolling
+  gz/gz_sensors_vendor:
+    type: git
+    url: https://github.com/gazebo-release/gz_sensors_vendor.git
+    version: rolling
+  gz/gz_msgs_vendor:
+    type: git
+    url: https://github.com/gazebo-release/gz_msgs_vendor.git
+    version: rolling
+  gz/gz_common_vendor:
+    type: git
+    url: https://github.com/gazebo-release/gz_common_vendor.git
+    version: rolling
+  gz/gz_rendering_vendor:
+    type: git
+    url: https://github.com/gazebo-release/gz_rendering_vendor.git
+    version: rolling
+  gz/gz_ogre_next_vendor:
+    type: git
+    url: https://github.com/gazebo-release/gz_ogre_next_vendor.git
+    version: rolling
+  gz/gz_physics_vendor:
+    type: git
+    url: https://github.com/gazebo-release/gz_physics_vendor.git
+    version: rolling
+  gz/gz_fuel_tools_vendor:
+    type: git
+    url: https://github.com/gazebo-release/gz_fuel_tools_vendor.git
+    version: rolling
+  gz/gz_gui_vendor:
+    type: git
+    url: https://github.com/gazebo-release/gz_gui_vendor.git
+    version: rolling
+  gz/gz_dartsim_vendor:
+    type: git
+    url: https://github.com/gazebo-release/gz_dartsim_vendor.git
+    version: rolling
```

Comment:
I think you'll want to include https://github.com/gazebosim/ros_gz/  because it also needs to be built with the version of Gazebo used here and `gz_ros2_control_demos` depends on it.

From the logs, it looks like you're actually still using Fortress

```
2024-05-14T13:32:36.3189979Z 9: test_arm (gz_ros2_control_tests.TestFixture) ... [INFO] [ruby $(which ign) gazebo-1]: process started with pid [87741]
2024-05-14T13:32:36.3190753Z 9: [INFO] [create-2]: process started with pid [87743]
2024-05-14T13:32:36.3191362Z 9: [INFO] [robot_state_publisher-3]: process started with pid [87745]
2024-05-14T13:32:36.3191932Z 9: [INFO] [python3-4]: process started with pid [87747]
2024-05-14T13:32:36.3192797Z 9: [robot_state_publisher-3] [INFO] [1715693451.304287286] [robot_state_publisher]: got segment cart
2024-05-14T13:32:36.3194370Z 9: [robot_state_publisher-3] [INFO] [1715693451.304382593] [robot_state_publisher]: got segment slideBar
2024-05-14T13:32:36.3195930Z 9: [robot_state_publisher-3] [INFO] [1715693451.304394776] [robot_state_publisher]: got segment world
2024-05-14T13:32:36.3197397Z 9: [create-2] [INFO] [1715693451.326267146] [ros_gz_sim]: Requesting list of world names.
2024-05-14T13:32:36.3198299Z 9: [create-2] [INFO] [1715693451.796072830] [ros_gz_sim]: Requested creation of entity.
2024-05-14T13:32:36.3199012Z 9: [create-2] [INFO] [1715693451.796125348] [ros_gz_sim]: OK creation of entity.
2024-05-14T13:32:36.3200040Z 9: [INFO] [create-2]: process has finished cleanly [pid 87743]
2024-05-14T13:32:36.3200936Z 9: [INFO] [ros2-5]: process started with pid [87795]
2024-05-14T13:32:36.3202448Z 9: [ruby $(which ign) gazebo-1] [Msg] Ignition Gazebo Server v6.16.0
2024-05-14T13:32:36.3204526Z 9: [ruby $(which ign) gazebo-1] [Msg] Loading SDF world file[/usr/share/ignition/ignition-gazebo6/worlds/empty.sdf].
2024-05-14T13:32:36.3206871Z 9: [ruby $(which ign) gazebo-1] [Msg] Serving entity system service on [/entity/system/add]

```

Note the version `2024-05-14T13:32:36.3202448Z 9: [ruby $(which ign) gazebo-1] [Msg] Ignition Gazebo Server v6.16.0`. Harmonic is 8.x

Reply:
I see, rosdep installs ros-iron-ros-gz-sim for example, which uses ignition-gazebo6-dev (fortress) and the rolling branch of gazebo_ros2_control is not working with it any more?